runtime/platform.c

```diff
@@ -115,6 +139,15 @@ void caml_plat_wait(caml_plat_cond* cond)
   check_err("wait", pthread_cond_wait(&cond->cond, cond->mutex));
 }
 
+int caml_plat_timedwait(caml_plat_cond* cond, const struct timespec *deadline)
+{
+  int res;
+  caml_plat_assert_locked(cond->mutex);
+  res = pthread_cond_timedwait(&cond->cond, cond->mutex, deadline);
```

Comment:
Are you sure this is correct? The condition var has an `&`, but the mutex doesn't.